apps/communications/contacts/test/marionette/list_test.js

```diff
@@ -51,6 +51,68 @@ marionette('Contacts > List', function() {
     });
   });
 
+  suite('List Order > ', function() {
+    var data = [['GG', 'E'], ['AA', 'Z'],
+      ['XX', 'C'], ['CC', 'X'], ['EE', 'G'], ['FF', 'F'], ['HH', 'D'],
+      ['BB', 'Y'], ['YY', 'B'], ['ZZ', 'A'], ['DD', 'H']];
+    setup(function() {
+      data.forEach(contactData => {
+        subject.addContact({
```

Comment:
As we're in an integration test (as opposed to end-to-end), I'd perform an injection of the contacts. Let's create a file called `inject.js`:
```js
injectContacts: function(contactsData) {
  contacts.forEach(function(contactData) {
    this.injectContact(contactData)
  })
},

injectContact: function(contactData) {
  this.client.executeAsyncScript(function(){
    var contact = new mozContact(contactData);
    var req = window.navigator.mozContacts.save(contact);
    req.onsuccess = function() {
      console.log('success saving contact');
      marionetteScriptFinished(true);
    };
    req.onerror = function() {
      console.error('error saving contact', req.error.name);
      marionetteScriptFinished(false);
    };

  }, [contactData]);
}
```

[Source](https://github.com/mozilla-b2g/gaia/blob/master/tests/atoms/gaia_data_layer.js#L67)